invenio_rdm_records/contrib/journal/custom_fields.py

```diff
@@ -12,13 +12,151 @@
 - journal.volume
 """
 
+from abc import ABC
+
 from invenio_i18n import lazy_gettext as _
 from invenio_records_resources.services.custom_fields import BaseCF
-from invenio_records_resources.services.records.facets import CFTermsFacet
-from marshmallow import fields
+from marshmallow import fields, missing
 from marshmallow_utils.fields import SanitizedUnicode
 
 
+class DumpProcessorMixin(ABC):
+    """Mixin class that defines pre_dump and post_dump methods to extend records serialization."""
+
+    def post_dump(self, data, original={}, **kwargs):
+        """Hook called after the marshmallow serialization of the record.
+
+        :param data: The dumped record data.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The serialized record data.
+        """
+        return data
+
+    def pre_dump(self, data, original={}, **kwargs):
+        """
+        Hook called before the marshmallow serialization of the record.
+
+        :param data: The record data to dump.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The data to dump.
+        """
+        return data
+
+
+class JournalDataciteProcessor(DumpProcessorMixin):
+    """Dump processor for datacite serialization of 'Journal' custom field."""
+
+    def post_dump(self, data, original={}, **kwargs):
```

Comment:
same, check all ocurrences of `={}` there might be more

Reply:
```suggestion
    def post_dump(self, data, original=None, **kwargs):
```